libs/database/src/lib/services/content.service.ts

```diff
@@ -1294,4 +1301,59 @@ Message: ${message}`,
       items: result,
     };
   };
+
+  /**
+   *
+   * @param {string} userId
+   * @param {number} maxResults
+   * @param {string} sinceId
+   * @param {string} untilId
+   * @returns
+   */
+  getEngagementFromUser = async (
+    userId: User,
+    sinceId: string,
+    untilId: string,
+    maxResults = DEFAULT_CONTENT_QUERY_OPTIONS.maxResults
+  ) => {
+    let filter: FilterQuery<Engagement> = {
+      'targetRef.$ref': 'content',
+      type: 'like',
+      user: userId,
+      visibility: EntityVisibility.Publish,
+    };
+    const totalDocument = await this._engagementModel
+      .countDocuments(filter)
+      .exec();
+
+    filter = await createCastcleFilter(filter, {
```

Comment:
```suggestion
    filter = createCastcleFilter(filter, {
```
`createCastcleFilter` is not a promise